### PR TITLE
better describe implications of disabling auto-sync, fixes #2009 [ready for second review]

### LINKED ISSUE #2009: Warning in disabling auto-sync is not warning that also quest download is now manual
Triggered by #1989 by @kudlav

"To upload your changes manually..." is not mentioning that also download is manually.

Probably it is better to add new text - to reduce workload for translators and to allow mentioning where "scan quests" is available.

**How to Reproduce**
Go to settings, switch to autosync (if disabled), switch to disabled autosync.

**Versions affected**
v22.1 on Android 10

### PR DESCRIPTION
**Pull request by matkoniecz**
Also, night mode display of this menu was fixed as a bonus (top icon is no longer black during night but is recolored to text color).

Thanks to @kudlav by reporting problem in #1989 and to @westnordost for hints.

Screenshot_2020-09-04-09-23-47-272_de westnordost streetcomplete debug

### CODE REVIEW

File: app/src/main/java/de/westnordost/streetcomplete/settings/SettingsFragment.kt [hunk 1]
```diff
@@ -111,9 +115,13 @@ class SettingsFragment : PreferenceFragmentCompat(),
             }
             Prefs.AUTOSYNC -> {
                 if (Prefs.Autosync.valueOf(prefs.getString(Prefs.AUTOSYNC, "ON")!!) != Prefs.Autosync.ON) {
+                    val view = LayoutInflater.from(activity).inflate(R.layout.dialog_tutorial_upload, null)
+                    val filled = context!!.getString(R.string.action_download)
+                    val uploadExplanation = view.findViewById<MaterialTextView>(R.id.tutorialDownloadPanel)
+                    uploadExplanation.text = context!!.getString(R.string.dialog_tutorial_download, filled)
```
Comment (westnordost) [inline app/src/main/java/de/westnordost/streetcomplete/settings/SettingsFragment.kt:121]: better use requireContext()

File: app/src/main/java/de/westnordost/streetcomplete/settings/SettingsFragment.kt [hunk 2]
```diff
@@ -111,9 +115,13 @@ class SettingsFragment : PreferenceFragmentCompat(),
             }
             Prefs.AUTOSYNC -> {
                 if (Prefs.Autosync.valueOf(prefs.getString(Prefs.AUTOSYNC, "ON")!!) != Prefs.Autosync.ON) {
+                    val view = LayoutInflater.from(activity).inflate(R.layout.dialog_tutorial_upload, null)
+                    val filled = context!!.getString(R.string.action_download)
+                    val uploadExplanation = view.findViewById<MaterialTextView>(R.id.tutorialDownloadPanel)
```
Comment (westnordost) [inline app/src/main/java/de/westnordost/streetcomplete/settings/SettingsFragment.kt:120]: MaterialTextView? It think it is a normal text view

File: app/src/main/res/layout/dialog_tutorial_upload.xml [hunk 3]
```diff
@@ -1,14 +1,28 @@
...
         android:drawablePadding="8dp"
+        app:drawableTint="@color/text"
         android:text="@string/dialog_tutorial_upload"
         android:textAppearance="@android:style/TextAppearance.Theme.Dialog"/>
-
-</RelativeLayout>
+    <View
+        android:layout_width="match_parent"
+        android:layout_height="3dp"
+        />
```
Comment (westnordost) [inline app/src/main/res/layout/dialog_tutorial_upload.xml:19]: Not sure how it looks, to nicely separate the two, you could also use a horizontal line (background color of view to `@color/divider`, layout height to 1dp + padding top and bottom)

File: app/src/main/res/drawable/ic_search_48dp.xml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+        android:width="48dp"
+        android:height="48dp"
+        android:viewportWidth="24.0"
+        android:viewportHeight="24.0">
+    <path
+        android:fillColor="#000"
```
Comment (westnordost) [inline app/src/main/res/drawable/ic_search_48dp.xml:7]: You should create a copy of this with `fillColor='#fff'` for night mode

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -891,5 +891,6 @@ Otherwise, you can download another keyboard in the app store. Popular keyboards
     <string name="at_housenumber">house number %s:</string>
     <string name="quest_summit_register_title">Is there a summit register at %s?</string>
     <string name="quest_is_defibrillator_inside_title">Is this defibrillator (AED) inside a building?</string>
+    <string name="dialog_tutorial_download">Quest download is also affected by auto-sync option. Use \"%s\" button in the menu to trigger download of quests.</string>
```
Comment (westnordost) [inline app/src/main/res/values/strings.xml:894]:
```suggestion
    <string name="dialog_tutorial_download">With auto-sync off, quests are not downloaded automatically. To do this manually, use the \"%s\" button in the menu at the location you want to download.</string>
```

**Review by westnordost — CHANGES_REQUESTED** [review]
x

File: app/src/main/res/layout/dialog_tutorial_upload.xml (hunk 3 above)
Comment (matkoniecz) [reply app/src/main/res/layout/dialog_tutorial_upload.xml:19]: I deliberately decided to not use line, as for me it is not about two strongly separate things, so strong separation was deliberately not used.

File: app/src/main/res/drawable/ic_search_48dp.xml (hunk 4 above)
Comment (matkoniecz) [reply app/src/main/res/drawable/ic_search_48dp.xml:7]: Oh right, unlike upload icon that was special (black also during night) here separate night icon would be better.

File: app/src/main/java/de/westnordost/streetcomplete/settings/SettingsFragment.kt (hunk 2 above)
Comment (matkoniecz) [reply app/src/main/java/de/westnordost/streetcomplete/settings/SettingsFragment.kt:120]: Weird, I added it to avoid crash. Removed.

**Review by westnordost — APPROVED** [review]

**Comment by matkoniecz** [comment]
@westnordost It seems to be reverted in merging of new quest downloader. Is it intentional? Is the new quest downloader now ignoring autosync setting?

**Comment by westnordost** [comment]
Yes, it is. The downloader ignores the autosync setting and the autosync setting has been renamed to only do stuff for uploading. I figured this makes more sense as there is little reason to download "by hand" anymore plus downloading of the map tiles happens anyway always

**Comment by timothywashere** [comment]
Personally I find that quite annoying.
I have PAYG mobile data and when I turn my mobile data on whilst using StreetComplete, it's usually to download the background map tiles so I can solve quests I have already downloaded. I don't want it to automatically download new quests as I am paying 1p per MB.  I would rather control it manually.
But I'm sure I'm a small minority, probably of 1 :)

**Comment by kudlav** [comment]
Hi, you are not alone. In the Czech Republic we have really expensive mobile data and low FUP limits. So it's no longer possible to use SC on 32kbps connection after the FUP has been reached.

The solution is simple: just download the area in advance using WiFi and then use the app completely offline.

**Comment by westnordost** [comment]
Back when I had a non-flatrate, I always had mobile data off except when I explicitly needed to use it.

**Pull request merged**